baseplate/sidecars/live_data_watcher.py

```diff
@@ -32,6 +37,64 @@ def __init__(self, dest: str, owner: int, group: int, mode: int):
         self.group = group
         self.mode = mode
 
+    @staticmethod
+    def get_encrypted_json_from_s3(
+        bucket_name: str, file_key: str, region_name: str, sse_key: str
+    ) -> Optional[bytes]:
+        s3_client = boto3.client(
+            "s3",
+            region_name=region_name,
+        )
+        data: Optional[bytes] = None
+        # The S3 data must be Server Side Encrypted and we should have the decryption key.
+        kwargs = {
+            "Bucket": bucket_name,
+            "Key": file_key,
+            "SSECustomerKey": sse_key,
+            "SSECustomerAlgorithm": "AES256",
+        }
+        try:
+            s3_object = s3_client.get_object(**kwargs)
+            # Returns bytes.
+            data = s3_object["Body"].read()
+        except ClientError as error:
+            logger.exception(
+                "Failed to load from S3. Received error code %s: %s",
+                error.response["Error"]["Code"],
+                error.response["Error"]["Message"],
+            )
+            return None
+        except ValueError as error:
+            logger.exception(error)
+            return None
+        return data
+
+    @staticmethod
+    def get_data_to_write(json_data: dict, data: bytes) -> Optional[bytes]:
+        # Check if we have a JSON in a special format containing:
+        # data = {
+        #    "live_data_watcher_load_type": str
+        # }
+        # If the load type is 'S3', this format is an indication that we support
+        # downloading the contents of encrypted files uploaded to S3.
+        if json_data.get("live_data_watcher_load_type") == "S3":
+            # Only write the data if we actually managed to fetch its contents.
+            bucket_name = json_data.get("bucket_name")
+            file_key = json_data.get("file_key")
+            sse_key = json_data.get("sse_key")
+            region_name = json_data.get("region_name")
+            # We require all of these keys to properly read from S3.
+            if bucket_name is None or file_key is None or sse_key is None or region_name is None:
+                logger.debug("Missing data in live data watch zk node to read from S3.")
+                return None
```

Comment:
instead of `return None`, should we fallback to just return the raw data to the caller?

Reply:
Felt weird for me to do that since we specifically expected it to be in this S3 JSON format.

If we return back this format as raw data, it seems buggy to me. I felt like doing "nothing" was better than setting the tmpfile to something that didn't make sense. Doing nothing would at least preserve the data from before?

Let me know what you think

Reply:
this is basically how confident do you think this is a json file intended to be comply to this spec just wrote wrong, vs. this is some file totally not intended to be used this way just happened to have `"live_data_watcher_load_type": "S3"` in it so `json.loads` thought it's valid json and treated it this way.

Reply:
Got it. I feel like "live_data_watcher_load_type": "S3" being in a JSON feels extremely unlikely so it feels like it would be a spec just wrong wrong.

I don't feel strongly about this though - if you feel like that's possible, I think returning the raw data is fine 👯 

Let me know what you prefer overall

Reply:
IMO either way we're gonna find an edge case that triggers the bug in one direction or the other - I personally agree with Sid that we should error out and make it clear that something has to change rather than silently passing data through